src/libmongoc/tests/test-mongoc-client-session.c

```diff
@@ -799,7 +820,7 @@ _test_end_sessions_many (bool pooled)
    endsessions_test_get_ended_lsids (&test, 0, &ended_lsids);
    ASSERT_CMPINT (bson_count_keys (&ended_lsids), ==, 10000);
    endsessions_test_get_ended_lsids (&test, 1, &ended_lsids);
-   ASSERT_CMPINT (bson_count_keys (&ended_lsids), ==, 1);
+   ASSERT_CMPINT (bson_count_keys (&ended_lsids), ==, 2);
```

Comment:
I do not think this assertion should change. This is asserting that the second "endSessions" command ends one session. AFAICT there should still be 10001 server sessions on the pool by the end of this test, not 10002. Modifying `send_ping` above may resolve that.